### PR TITLE
Plot optimization

### PR DESCRIPTION
**Pull request by gdiazc**
The Elasticsearch instance now stores pre-computed "mean hourly sentiment ticks" in a separate index. This pull requests migrates the dashboard over to this pre-computed data so it doesn't have to be calculated on the fly.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -98,5 +98,28 @@ def get_tweets():
     return json.dumps(ans)
 
 
+@app.route('/get_mean_sentiments.json', methods=['GET'])
+def get_mean_sentiments():
```
Comment (dev1-fund3) [inline app.py:62]: add docstring comments for what this method does

File: es_client.py [hunk 2]
```diff
@@ -70,3 +69,32 @@ def get_tweets(client, max_tweets=500):
                 tweets.append(t)
 
     return tweets
+
+
+def get_mean_hourly_sentiment_ticks(client, max_ticks=1000):
+    """
+    Get mean hourly sentiment ticks from the Elasticsearch endpoint specified by client.
+
+    :param client: Elasticsearch client object.
```
Comment (dev1-fund3) [inline es_client.py:86]: add docstring for max_ticks

File: es_client.py [hunk 3]
```diff
@@ -70,3 +69,32 @@ def get_tweets(client, max_tweets=500):
...
+    """
+
+    # TODO: match a specific version
+    search = Search(index='mean_sentiment_ticks')\
+        .using(client)\
+        .query('match_all')\
+        .sort({'timestamp': {'order': 'desc'}})[:max_ticks]
+    search.execute()
+
+    ticks = []
```
Comment (dev1-fund3) [inline es_client.py]:
```suggestion
    return [
        {
            'timestamp': hit.to_dict().get('timestamp', None),
            'value': hit.to_dict().get('value', None)
        } for hit in search
    ]
```

Comment (dev1-fund3) [reply es_client.py]: Use list compression and fewer local vars for speed up.

File: plotting.py [hunk 4]
```diff
@@ -33,6 +35,8 @@ def plot_polarity_vs_time(df, polarity_column='polarity'):
     p.xaxis.formatter = bokeh.models.DatetimeTickFormatter(hours=['%m/%d %H:%M'])
 
     # Format view range:
+    now = datetime.datetime.now()
+    p.x_range = bokeh.models.Range1d(now - datetime.timedelta(days=3), now + datetime.timedelta(hours=6))
```
Comment (dev1-fund3) [inline plotting.py:40]: add a low priority TODO to make the range able to be input by user via dropdown menu.

File: static/bitcoin_sentiments.js [hunk 5]
```diff
@@ -26,13 +32,22 @@ angular.module('BitcoinSentimentsApp', [])
                 {'id_str': 1, 'full_text': 'working!'}
             ]
 
-            BitcoinSentimentsService.getTweetsFromServer()
+            // Get sentiment ticks quicker:
```
Comment (dev1-fund3) [inline static/bitcoin_sentiments.js]:
```suggestion
            // Get precomputed sentiment ticks:
```

**Review by dev1-fund3 — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (gdiazc) [reply app.py:62]: Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

File: es_client.py (hunk 2 above)
Comment (gdiazc) [reply es_client.py:86]: Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

File: es_client.py (hunk 3 above)
Comment (gdiazc) [reply es_client.py]: Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

File: plotting.py (hunk 4 above)
Comment (gdiazc) [reply plotting.py:40]: Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

File: static/bitcoin_sentiments.js (hunk 5 above)
Comment (gdiazc) [reply static/bitcoin_sentiments.js]: Fixed in [3fd135b](https://github.com/fund3/sentiment-dashboard/pull/5/commits/3fd135bb0e6622d2d74fb1c882b41c9378f306c3)

**Review by dev1-fund3 — APPROVED** [review]

**Pull request merged**